platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/ProblemReportingCrossProjectModelAccess.kt

```diff
@@ -1045,9 +1040,70 @@ class ProblemReportingCrossProjectModelAccess(
             problems.onProblem(problem)
             coupledProjectsListener.onProjectReference(referrer.owner, delegate.owner)
             // Configure the target project, if it would normally be configured before the referring project
-            if (delegate.compareTo(referrer) < 0 && delegate.parent != null) {
+            if (delegate < referrer && delegate.parent != null && buildModelParameters.isInvalidateCoupledProjects) {
                 delegate.owner.ensureConfigured()
             }
         }
+
+        private
+        fun <T> DefaultProject.withEvaluationOrderCheck(action: DefaultProject.() -> T?): T? {
+            // Attempt to get a result, since plugin might to contribute to DynamicObject properties
+            var failure: Throwable? = null
+            val result =
+                try {
+                    action()
+                } catch (e: Throwable) {
+                    failure = e
+                    null
+                }
+            if (failure == null) return result
+
+            return when {
+                // Referrer is configuring before the referent
+                this > referrer -> {
+                    reportEvaluationOrderProblem()
```

Comment:
I think we should report this regardless of the project ordering, and regardless of whether or not the operation succeeds.

It's also not really a separate problem, it's a suggestion for how to make build logic more reliable if the access cannot be easily removed. So, I'd merge it with the "project x cannot access project y" problem, and mention that declaring the dependency can be used as a work around if the access cannot be easily removed, rather than treating it as a separate problem.

Reply:
@adammurdoch should we report it only in `getProperty`/`invokeMethod`, or in each `onAccess` invocation?

Reply:
I guess only in `getProperty`/`invokeMethod` because the rest of the methods are not requiring for target be configured?